blockchain/state/statedb.go

```diff
@@ -816,36 +818,48 @@ func (self *StateDB) GetRefund() uint64 {
 
 // Finalise finalises the state by removing the self destructed objects
 // and clears the journal as well as the refunds.
-func (s *StateDB) Finalise(deleteEmptyObjects bool) {
-	for addr := range s.journal.dirties {
-		stateObject, exist := s.stateObjects[addr]
+func (stateDB *StateDB) Finalise(deleteEmptyObjects bool, setStorageRoot bool) {
+	for addr := range stateDB.journal.dirties {
+		stateObject, exist := stateDB.stateObjects[addr]
```

Comment:
How about change this like line 848.
```suggestion
		so, exist := stateDB.stateObjects[addr]
```

Reply:
Looks better!